### ISSUE TITLE
Style/StringChars cop throws an error when parsing String#split without parentheses

### ISSUE THREAD

**Issue by agargiulo** [rubocop/rubocop#9676]
The `Style/StringChars` cop throws an error when `String#split` is called without parentheses 

--------

## Expected behavior

The `Style/StringChars` cop should report an offense for the line

## Actual behavior

```
[21:21:20] % rubocop -d style_string_chars_test.rb
For /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq: configuration from /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/.rubocop.yml
Default configuration from /Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/config/default.yml
Inspecting 1 file
Scanning /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/style_string_chars_test.rb
An error occurred while Style/StringChars cop was inspecting /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/style_string_chars_test.rb:5:14.
undefined method `begin_pos' for nil:NilClass
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/parser-3.0.0.0/lib/parser/source/range.rb:211:in `join'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/style/string_chars.rb:29:in `on_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:136:in `public_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:166:in `with_cop_error_handling'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:134:in `each'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:158:in `on_numblock'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:71:in `on_numblock'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-ast-1.4.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/team.rb:157:in `investigate_partial'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:315:in `inspect_file'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:286:in `loop'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:132:in `block in file_offenses'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:157:in `file_offense_cache'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:131:in `file_offenses'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:122:in `process_file'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:100:in `each'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:100:in `reduce'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:100:in `each_inspected_file'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:86:in `inspect_files'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/runner.rb:47:in `run'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/command.rb:11:in `run'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli/environment.rb:18:in `run'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli.rb:65:in `run_command'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli.rb:72:in `execute_runners'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/lib/rubocop/cli.rb:41:in `run'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/exe/rubocop:13:in `block in <top (required)>'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
/Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/exe/rubocop:12:in `<top (required)>'
/Users/agargiulo/.rbenv/versions/3.0.0/bin/rubocop:23:in `load'
/Users/agargiulo/.rbenv/versions/3.0.0/bin/rubocop:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/StringChars cop was inspecting /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/style_string_chars_test.rb:5:14.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.12.1 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-darwin20)
Finished in 0.2596279999706894 seconds
```

## Steps to reproduce the problem

`style_string_chars_test.rb`:
```ruby
# frozen_string_literal: true

orig = %w[..##.#..#. ##..#..... #...##..#. ####.#...# ##.##.###. ##...#.### .#.#.#..## ..#....#.. ###...#.#. ..###..###]

pp orig.map { _1.split '' }
```
`.rubocop.yml`:
```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.0
```

create those two files then run `rubocop -d style_string_chars_test.rb`

If I change the line to `orig.map { _1.split('') }` then the error never happens and I get the correct offense registered
```
[21:28:01] % rubocop -d style_string_chars_test.rb
For /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq: configuration from /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/.rubocop.yml
Default configuration from /Users/agargiulo/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.1/config/default.yml
Inspecting 1 file
Scanning /private/var/folders/t3/mphyq08965q5blrx_zhfvrs40000gn/T/tmp.0qEGBoeq/style_string_chars_test.rb
C

Offenses:

style_string_chars_test.rb:5:18: C: [Correctable] Style/StringChars: Use chars instead of split('').
pp orig.map { _1.split('') }
                 ^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.17933900002390146 seconds
```

## RuboCop version

```
$ rubocop -V
1.12.1 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-darwin20)
```

**Comment by agargiulo** [comment]
I tried this just now without the numbered parameters and it still throws an error. 
```
# frozen_string_literal: true

orig = %w[..##.#..#. ##..#..... #...##..#. ####.#...# ##.##.###. ##...#.### .#.#.#..## ..#....#.. ###...#.#. ..###..###]

pp(orig.map { |foo| foo.split '' })
```

it's the same as above, ```undefined method `begin_pos' for nil:NilClass```
